python/cuml/internals/api_context_managers.py

```diff
@@ -153,33 +208,58 @@ def output_type(self, value: str):
 
     @property
     def memory_type(self):
+        """
+        MemoryType: The current memory type for the internal API context.
```

Comment:
Describe what memory type means and how it affects a call within this context manager.